src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/util/DelayedTaskScheduler.java

```diff
@@ -52,11 +51,12 @@ public int process()
         {
             long c = System.currentTimeMillis();
             SortedSet<Long> past = scheduledTimes.headSet(c);
-            Iterator<Long> iter = past.iterator();
-            while(iter.hasNext())
+
+            past = new TreeSet<>(past);
+            scheduledTimes.removeAll(past);
+
+            for (Long time : past)
```

Comment:
This is going to create a lot of garbage.

Reply:
creating new TreeSet adds some garbage.

I think we have some cases where a task being executed adds a new task, hence concurrent modification exception. We need some collection which is not backed by the original collection (or add needs to defer adding to the scheduledTimes when there is an iterator is running. At the end of the iterator, we add all the deferred tasks)

Reply:
Agreed. Recommend deferring the adds until after the iteration.